### PR TITLE
Fix precedence of Dot wrt Hash

### LINKED ISSUE #1042: Unecessary parens on method call
On `master`:

```ocaml
let _ = (x.foo)#m
```

Parentheses are necessary.

### PR DESCRIPTION
**Pull request by gpetiot**
Fix #1042

### CODE REVIEW

File: test/passing/index_op.ml [hunk 1]
```diff
@@ -113,3 +113,10 @@ let _ =
   []
 
 let _ = x.(y) <- (z.(w) <- u)
+
+let _ = x.foo#m
+
+class free =
+  object (m : 'test)
+    method get_def = m#state.def
```
Comment (gpetiot) [inline test/passing/index_op.ml]: the CI should fail here (I cleaned the cache)

**Comment by Julow** [comment]
No, `.` and `#` should have the same precedence. `a.b#c.d` is the same as `((a.b)#c).d`.

**Comment by gpetiot** [comment]
Are you sure about this? I used https://caml.inria.fr/pub/docs/manual-ocaml/expr.html#sec133 for reference but maybe I missed something.

**Comment by Julow** [comment]
After experimenting they seem to have the same precedence and associativity.

This 2 expressions parse to the same AST:
```ocaml
a.b#c.*(d)#e.(f);;
((((a.b)#c).*(d))#e).(f);;
```

and OCamlformat format them to `((a.b)#c.*(d))#e.(f)`

**Comment by gpetiot** [comment]
And are we sure this is not a parsing bug? I find it weird to not find any indication in the manual

**Comment by Julow** [comment]
It's not a bug but it's weird indeed.

Actually the table shoud look like this:

| | |
| --- | --- |
| `.`, `.(`, `.[`, `.{` | - |
| `#...` | left |
| `#id` | - |

The `#...` is not the method call syntax but HASHOP, custom operators starting with `#`.
`#id` is lower but parenthesis are not needed because there can only be an identifier on the right of the `#` (it has no associativity, like `.`, the parser will only look for an identifier on the right).

This cannot be solved by changing the precedence and needs a special case for `a#b.c`

**Comment by gpetiot** [comment]
I used the `Dot` precedence for `x#y`, here is the diff of test_branch, that looks better:

```
diff --git a/examples/boulderdash/boulderdash.ml b/examples/boulderdash/boulderdash.ml
index 2532c1659..4c1b0d720 100644
--- a/examples/boulderdash/boulderdash.ml
+++ b/examples/boulderdash/boulderdash.ml
@@ -105,7 +105,7 @@ let img_assoc v =
 
 let set_cell state x y v =
   state.map.(y).(x) <- v;
-  (state.imgs.(y).(x))##.src := img_assoc v
+  state.imgs.(y).(x)##.src := img_assoc v
 
 let walkable = function
   | Empty | Grass | Diamond | End -> true
@@ -151,9 +151,9 @@ let rec build_interaction state show_rem ((_, _, clock_stop) as clock) =
   >>= fun () ->
   for y = 0 to Array.length state.map - 1 do
     for x = 0 to Array.length state.map.(y) - 1 do
-      (state.imgs.(y).(x))##.onmouseover := Html.no_handler;
-      (state.imgs.(y).(x))##.onmouseout := Html.no_handler;
-      (state.imgs.(y).(x))##.onclick := Html.no_handler
+      state.imgs.(y).(x)##.onmouseover := Html.no_handler;
+      state.imgs.(y).(x)##.onmouseout := Html.no_handler;
+      state.imgs.(y).(x)##.onclick := Html.no_handler
     done
   done;
   let inhibit f _x =
@@ -184,12 +184,12 @@ let rec build_interaction state show_rem ((_, _, clock_stop) as clock) =
   let rec update (x, y) next img over_cont out_cont click_cont =
     if walkable state.map.(y).(x)
     then (
-      let cur_img = (state.imgs.(y).(x))##.src in
+      let cur_img = state.imgs.(y).(x)##.src in
       let over () =
-        (state.imgs.(y).(x))##.src := img;
+        state.imgs.(y).(x)##.src := img;
         over_cont ()
       and out () =
-        (state.imgs.(y).(x))##.src := cur_img;
+        state.imgs.(y).(x)##.src := cur_img;
         out_cont ()
       and click' () =
         click_cont ()
@@ -222,11 +222,11 @@ let rec build_interaction state show_rem ((_, _, clock_stop) as clock) =
             | _ -> Lwt.fail e)
         >>= fun () -> build_interaction state show_rem clock
       in
-      (state.imgs.(y).(x))##.onmouseover
+      state.imgs.(y).(x)##.onmouseover
       := Html.handler (inhibit (set_pending_out (with_pending_out over) out));
-      (state.imgs.(y).(x))##.onmouseout
+      state.imgs.(y).(x)##.onmouseout
       := Html.handler (inhibit (with_pending_out (fun () -> Lwt.return ())));
-      (state.imgs.(y).(x))##.onclick := Html.handler (inhibit (with_pending_out click));
+      state.imgs.(y).(x)##.onclick := Html.handler (inhibit (with_pending_out click));
       if state.map.(y).(x) <> End then update (next (x, y)) next img over out click')
   in
   let update_push ((x, y) as pos) next img img_guy =
@@ -236,13 +236,13 @@ let rec build_interaction state show_rem ((_, _, clock_stop) as clock) =
        with Invalid_argument _ -> false
     then (
       let over () =
-        (state.imgs.(y).(x))##.src := img_guy;
-        (state.imgs.(y').(x'))##.src := img;
+        state.imgs.(y).(x)##.src := img_guy;
+        state.imgs.(y').(x')##.src := img;
         Lwt.return ()
       in
       let out () =
-        (state.imgs.(y).(x))##.src := js "sprites/guy.png";
-        (state.imgs.(y').(x'))##.src := js "sprites/boulder.png"
+        state.imgs.(y).(x)##.src := js "sprites/guy.png";
+        state.imgs.(y').(x')##.src := js "sprites/boulder.png"
       in
       let click () =
         set_cell state x y Empty;
@@ -259,11 +259,11 @@ let rec build_interaction state show_rem ((_, _, clock_stop) as clock) =
             | e -> Lwt.fail e)
         >>= fun () -> build_interaction state show_rem clock
       in
-      (state.imgs.(y').(x'))##.onmouseover
+      state.imgs.(y').(x')##.onmouseover
       := Html.handler (inhibit (set_pending_out (with_pending_out over) out));
-      (state.imgs.(y').(x'))##.onmouseout
+      state.imgs.(y').(x')##.onmouseout
       := Html.handler (inhibit (with_pending_out (fun () -> Lwt.return ())));
-      (state.imgs.(y').(x'))##.onclick := Html.handler (inhibit (with_pending_out click)))
+      state.imgs.(y').(x')##.onclick := Html.handler (inhibit (with_pending_out click)))
   in
   if state.pos = state.endpos
   then (
@@ -277,7 +277,7 @@ let rec build_interaction state show_rem ((_, _, clock_stop) as clock) =
     if state.rem = 0
     then (
       let x, y = state.endpos in
-      (state.imgs.(y).(x))##.src := js "sprites/end.png";
+      state.imgs.(y).(x)##.src := js "sprites/end.png";
       state.map.(y).(x) <- End);
     let r (x, y) = succ x, y and l (x, y) = pred x, y in
     let u (x, y) = x, pred y and d (x, y) = x, succ y in
diff --git a/examples/graph_viewer/svg.ml b/examples/graph_viewer/svg.ml
index 063f3e24f..ee1ef0262 100644
--- a/examples/graph_viewer/svg.ml
+++ b/examples/graph_viewer/svg.ml
@@ -487,7 +487,7 @@ let redraw w range ev =
   (*
   let t1 = Unix.gettimeofday () in
 *)
-  let ctx = Cairo_lablgtk.create (w#misc)#window in
+  let ctx = Cairo_lablgtk.create w#misc#window in
   Cairo.save ctx;
   if !bboxes = [] then bboxes := List.map (fun e -> compute_extent ctx e) l;
   Cairo.new_path ctx;
@@ -495,7 +495,7 @@ let redraw w range ev =
   let rect = Gdk.Rectangle.create 0 0 0 0 in
   Gdk.Region.get_clipbox (GdkEvent.Expose.region ev) rect;
   Cairo.clip ctx;
-  let scale = scale *. ((1. /. scale) ** (range#adjustment)#value) in
+  let scale = scale *. ((1. /. scale) ** range#adjustment#value) in
   Cairo.scale ctx scale scale;
   Cairo.translate ctx 364. 22443.;
   let bbox =
@@ -519,8 +519,8 @@ Format.eprintf "%f %f %f %f (%f)@." x1 y1 x2 y2 scale;
   true
 
 let slider_changed (area : GMisc.drawing_area) range () =
-  let scale = scale *. ((1. /. scale) ** (range#adjustment)#value) in
-  (area#misc)#set_size_request
+  let scale = scale *. ((1. /. scale) ** range#adjustment#value) in
+  area#misc#set_size_request
     ~width:(truncate (width *. scale))
     ~height:(truncate (height *. scale))
     ();
@@ -530,7 +530,7 @@ let _ =
   ignore (GMain.Main.init ());
   let initial_size = 600 in
   let w = GWindow.window () in
-  ignore ((w#connect)#destroy GMain.quit);
+  ignore (w#connect#destroy GMain.quit);
   let b = GPack.vbox ~spacing:6 ~border_width:12 ~packing:w#add () in
   (*
   let f = GBin.frame ~shadow_type:`IN
@@ -548,12 +548,12 @@ let _ =
       ~packing:f#add_with_viewport
       ()
   in
-  (area#misc)#set_size_request
+  area#misc#set_size_request
     ~width:(truncate (width *. scale))
     ~height:(truncate (height *. scale))
     ();
   let slider = GRange.scale `HORIZONTAL ~draw_value:false ~packing:b#pack () in
-  (slider#adjustment)#set_bounds ~lower:0. ~upper:1. ~step_incr:0.1 ();
+  slider#adjustment#set_bounds ~lower:0. ~upper:1. ~step_incr:0.1 ();
   (*
   let button = GButton.check_button ~label:"Animate"
       ~packing:b#pack () in
@@ -565,8 +565,8 @@ let _ =
   ignore (button#connect#toggled
             (animate_toggled button slider)) ;
 *)
-  ignore (((area#event)#connect)#expose (redraw area slider));
-  ignore ((slider#connect)#value_changed (slider_changed area slider));
+  ignore (area#event#connect#expose (redraw area slider));
+  ignore (slider#connect#value_changed (slider_changed area slider));
   w#show ();
   GMain.main ()
 
diff --git a/examples/graph_viewer/viewer.ml b/examples/graph_viewer/viewer.ml
index 3bcd2d0e0..3a6970d1a 100644
--- a/examples/graph_viewer/viewer.ml
+++ b/examples/graph_viewer/viewer.ml
@@ -126,7 +126,7 @@ module Common = Viewer_common.F (struct
 
   type pixmap = GDraw.pixmap
 
-  let get_drawable w = new GDraw.drawable (w#misc)#window
+  let get_drawable w = new GDraw.drawable w#misc#window
 
   let make_pixmap window width height = GDraw.pixmap ~width ~height ~window ()
 
@@ -154,9 +154,9 @@ open Common
 
 let set_visible w vis =
   if vis
-  then (if not (w#misc)#visible then (w#misc)#show ())
-  else if (w#misc)#visible
-  then (w#misc)#hide ()
+  then (if not w#misc#visible then w#misc#show ())
+  else if w#misc#visible
+  then w#misc#hide ()
   (*
   let button = GButton.check_button ~label:"Animate"
       ~packing:b#pack () in
@@ -565,8 +565,8 @@ let _ =
   ignore (button#connect#toggled
             (animate_toggled button slider)) ;
 *)
-  ignore (((area#event)#connect)#expose (redraw area slider));
-  ignore ((slider#connect)#value_changed (slider_changed area slider));
+  ignore (area#event#connect#expose (redraw area slider));
+  ignore (slider#connect#value_changed (slider_changed area slider));
   w#show ();
   GMain.main ()
 
diff --git a/examples/graph_viewer/viewer.ml b/examples/graph_viewer/viewer.ml
index 3bcd2d0e0..3a6970d1a 100644
--- a/examples/graph_viewer/viewer.ml
+++ b/examples/graph_viewer/viewer.ml
@@ -126,7 +126,7 @@ module Common = Viewer_common.F (struct
 
   type pixmap = GDraw.pixmap
 
-  let get_drawable w = new GDraw.drawable (w#misc)#window
+  let get_drawable w = new GDraw.drawable w#misc#window
 
   let make_pixmap window width height = GDraw.pixmap ~width ~height ~window ()
 
@@ -154,9 +154,9 @@ open Common
 
 let set_visible w vis =
   if vis
-  then (if not (w#misc)#visible then (w#misc)#show ())
-  else if (w#misc)#visible
-  then (w#misc)#hide ()
+  then (if not w#misc#visible then w#misc#show ())
+  else if w#misc#visible
+  then w#misc#hide ()
 
 let scroll_view ?width ?height ?packing st =
   let table = GPack.table ?width ?height ~columns:2 ~rows:2 ?packing () in
@@ -168,7 +168,7 @@ let scroll_view ?width ?height ?packing st =
       ~packing:(table#attach ~left:0 ~top:1 ~fill:`BOTH ~expand:`NONE)
       ()
   in
-  (hbar#misc)#hide ();
+  hbar#misc#hide ();
   let vadj = GData.adjustment () in
   let vbar =
     GRange.scrollbar
@@ -177,12 +177,12 @@ let scroll_view ?width ?height ?packing st =
       ~packing:(table#attach ~left:1 ~top:0 ~fill:`BOTH ~expand:`NONE)
       ()
   in
-  (vbar#misc)#hide ();
+  vbar#misc#hide ();
   let display =
     GMisc.drawing_area ~packing:(table#attach ~left:0 ~top:0 ~fill:`BOTH ~expand:`BOTH) ()
   in
-  (display#misc)#set_can_focus true;
-  (display#misc)#set_double_buffered false;
+  display#misc#set_can_focus true;
+  display#misc#set_double_buffered false;
   let sadj = GData.adjustment ~upper:20. ~step_incr:1. ~page_incr:1. ~page_size:0. () in
   let zoom_steps = 8. in
   (* Number of steps to get a factor of 2 *)
@@ -195,7 +195,7 @@ let scroll_view ?width ?height ?packing st =
   in
   let get_scale () = (2. ** (sadj#value /. zoom_steps)) /. st.zoom_factor in
   let update_scrollbars () =
-    let a = (display#misc)#allocation in
+    let a = display#misc#allocation in
     let scale = get_scale () in
     let aw = ceil (float a.Gtk.width /. scale) in
     let ah = ceil (float a.Gtk.height /. scale) in
@@ -223,13 +223,13 @@ let scroll_view ?width ?height ?packing st =
     GtkBase.Widget.queue_draw display#as_widget
   in
   ignore
-    (((display#event)#connect)#configure (fun ev ->
+    (display#event#connect#configure (fun ev ->
          prerr_endline "CONFIGURE";
          update_scrollbars ();
          false));
   ignore
-    (((display#event)#connect)#map (fun ev ->
-         let a = (display#misc)#allocation in
+    (display#event#connect#map (fun ev ->
+         let a = display#misc#allocation in
          Format.eprintf "alloc: %d %d@." a.Gtk.width a.Gtk.height;
          let zoom_factor =
            max (st.st_width /. float a.Gtk.width) (st.st_height /. float a.Gtk.height)
@@ -238,9 +238,9 @@ let scroll_view ?width ?height ?packing st =
          refresh ();
          update_scrollbars ();
          false));
-  (display#event)#add [`STRUCTURE];
+  display#event#add [`STRUCTURE];
   ignore
-    (((display#event)#connect)#expose (fun ev ->
+    (display#event#connect#expose (fun ev ->
          let area = GdkEvent.Expose.area ev in
          let x = Gdk.Rectangle.x area in
          let y = Gdk.Rectangle.y area in
@@ -252,20 +252,20 @@ let scroll_view ?width ?height ?packing st =
            hadj#value
            vadj#value
            display
-           (display#misc)#allocation
+           display#misc#allocation
            x
            y
            width
            height;
          true));
   ignore
-    ((hadj#connect)#value_changed (fun () -> GtkBase.Widget.queue_draw display#as_widget));
+    (hadj#connect#value_changed (fun () -> GtkBase.Widget.queue_draw display#as_widget));
   ignore
-    ((vadj#connect)#value_changed (fun () -> GtkBase.Widget.queue_draw display#as_widget));
+    (vadj#connect#value_changed (fun () -> GtkBase.Widget.queue_draw display#as_widget));
   let prev_scale = ref (get_scale ()) in
   let zoom_center = ref (0.5, 0.5) in
   ignore
-    ((sadj#connect)#value_changed (fun () ->
+    (sadj#connect#value_changed (fun () ->
          let scale = get_scale () in
          let r = 1. -. (!prev_scale /. scale) in
          Format.eprintf "update@.";
@@ -275,7 +275,7 @@ let scroll_view ?width ?height ?packing st =
          refresh ();
          update_scrollbars ()));
   let bump_scale x y v =
-    let a = (display#misc)#allocation in
+    let a = display#misc#allocation in
     let x = x /. float a.Gtk.width in
     let y = y /. float a.Gtk.height in
     if x >= 0. && x <= 1. && y >= 0. && y <= 1. then zoom_center := x, y;
@@ -287,33 +287,33 @@ let scroll_view ?width ?height ?packing st =
   in
   (* Zoom using the mouse wheel *)
   ignore
-    (((display#event)#connect)#scroll (fun ev ->
+    (display#event#connect#scroll (fun ev ->
          let x = GdkEvent.Scroll.x ev in
          let y = GdkEvent.Scroll.y ev in
          match GdkEvent.Scroll.direction ev with
          | `UP -> bump_scale x y 1.
          | `DOWN -> bump_scale x y (-1.)
            width
            height;
          true));
   ignore
-    ((hadj#connect)#value_changed (fun () -> GtkBase.Widget.queue_draw display#as_widget));
+    (hadj#connect#value_changed (fun () -> GtkBase.Widget.queue_draw display#as_widget));
   ignore
-    ((vadj#connect)#value_changed (fun () -> GtkBase.Widget.queue_draw display#as_widget));
+    (vadj#connect#value_changed (fun () -> GtkBase.Widget.queue_draw display#as_widget));
   let prev_scale = ref (get_scale ()) in
   let zoom_center = ref (0.5, 0.5) in
   ignore
-    ((sadj#connect)#value_changed (fun () ->
+    (sadj#connect#value_changed (fun () ->
          let scale = get_scale () in
          let r = 1. -. (!prev_scale /. scale) in
          Format.eprintf "update@.";
@@ -275,7 +275,7 @@ let scroll_view ?width ?height ?packing st =
          refresh ();
          update_scrollbars ()));
   let bump_scale x y v =
-    let a = (display#misc)#allocation in
+    let a = display#misc#allocation in
     let x = x /. float a.Gtk.width in
     let y = y /. float a.Gtk.height in
     if x >= 0. && x <= 1. && y >= 0. && y <= 1. then zoom_center := x, y;
@@ -287,33 +287,33 @@ let scroll_view ?width ?height ?packing st =
   in
   (* Zoom using the mouse wheel *)
   ignore
-    (((display#event)#connect)#scroll (fun ev ->
+    (display#event#connect#scroll (fun ev ->
          let x = GdkEvent.Scroll.x ev in
          let y = GdkEvent.Scroll.y ev in
          match GdkEvent.Scroll.direction ev with
          | `UP -> bump_scale x y 1.
          | `DOWN -> bump_scale x y (-1.)
          | _ -> false));
-  (display#event)#add [`SCROLL];
+  display#event#add [`SCROLL];
   let pos = ref None in
   ignore
-    (((display#event)#connect)#button_press (fun ev ->
-         (display#misc)#grab_focus ();
+    (display#event#connect#button_press (fun ev ->
+         display#misc#grab_focus ();
          if GdkEvent.get_type ev = `BUTTON_PRESS && GdkEvent.Button.button ev = 1
          then pos := Some (GdkEvent.Button.x ev, GdkEvent.Button.y ev);
          false));
   ignore
-    (((display#event)#connect)#button_release (fun ev ->
+    (display#event#connect#button_release (fun ev ->
          if GdkEvent.Button.button ev = 1 then pos := None;
          false));
   ignore
-    (((display#event)#connect)#motion_notify (fun ev ->
+    (display#event#connect#motion_notify (fun ev ->
          (match !pos with
          | Some (x, y) ->
              let x', y' =
                if GdkEvent.Motion.is_hint ev
                then
-                 let x', y' = (display#misc)#pointer in
+                 let x', y' = display#misc#pointer in
                  float x', float y'
                else GdkEvent.Motion.x ev, GdkEvent.Motion.y ev
              in
@@ -324,10 +324,9 @@ let scroll_view ?width ?height ?packing st =
              pos := Some (x', y')
          | None -> ());
          false));
-  (display#event)#add
-    [`BUTTON_PRESS; `BUTTON_RELEASE; `BUTTON1_MOTION; `POINTER_MOTION_HINT];
+  display#event#add [`BUTTON_PRESS; `BUTTON_RELEASE; `BUTTON1_MOTION; `POINTER_MOTION_HINT];
   ignore
-    (((display#event)#connect)#key_press (fun ev ->
+    (display#event#connect#key_press (fun ev ->
          let keyval = GdkEvent.Key.keyval ev in
          if keyval = GdkKeysyms._Up
          then (
@@ -361,7 +360,7 @@ let scroll_view ?width ?height ?packing st =
            true)
          else if keyval = GdkKeysyms._0 || keyval = GdkKeysyms._agrave
          then (
-           let a = (table#misc)#allocation in
+           let a = table#misc#allocation in
            Format.eprintf "alloc: %d %d@." a.Gtk.width a.Gtk.height;
            let zf =
              max (st.st_width /. float a.Gtk.width) (st.st_height /. float a.Gtk.height)
@@ -378,14 +377,14 @@ let scroll_view ?width ?height ?packing st =
                  || keyval = GdkKeysyms._equal
                  || keyval = GdkKeysyms._KP_Add
          then
-           let x, y = (display#misc)#pointer in
+           let x, y = display#misc#pointer in
            bump_scale (float x) (float y) 1.
          else if keyval = GdkKeysyms._minus || keyval = GdkKeysyms._KP_Subtract
          then
-           let x, y = (display#misc)#pointer in
+           let x, y = display#misc#pointer in
            bump_scale (float x) (float y) (-1.)
          else false));
-  (display#event)#add [`KEY_PRESS];
+  display#event#add [`KEY_PRESS];
   object
     method scale_adjustment = sadj
   end
@@ -403,7 +402,7 @@ let create ?(full_screen = false) (x1, y1, x2, y2) scene =
   in
   let initial_size = 600 in
   let w = GWindow.window () in
-  ignore ((w#connect)#destroy GMain.quit);
+  ignore (w#connect#destroy GMain.quit);
   let b = GPack.hbox ~packing:w#add () in
   let f =
     scroll_view ~width:initial_size ~height:initial_size ~packing:(b#pack ~expand:true) st
@@ -431,7 +430,7 @@ Format.eprintf "%d %d %b@." x y kbd; false));
   in
   if full_screen then ignore (toggle_fullscreen ());
   ignore
-    (((w#event)#connect)#key_press (fun ev ->
+    (w#event#connect#key_press (fun ev ->
          let keyval = GdkEvent.Key.keyval ev in
          if keyval = GdkKeysyms._q || keyval = GdkKeysyms._Q
          then exit 0
diff --git a/examples/minesweeper/minesweeper.ml b/examples/minesweeper/minesweeper.ml
index d58ad8da0..5bc984421 100644
--- a/examples/minesweeper/minesweeper.ml
+++ b/examples/minesweeper/minesweeper.ml
@@ -164,7 +164,7 @@ let draw_board d =
 let disable_events d =
   for y = 0 to d.cf.nbrows - 1 do
     for x = 0 to d.cf.nbcols - 1 do
-      (d.dom.(y).(x))##.onclick
+      d.dom.(y).(x)##.onclick
       := Html.handler (fun _ ->
              Html.window##alert (js "GAME OVER");
              Js._false)
```

File: src/Ast.ml [hunk 2]
```diff
@@ -1662,7 +1662,7 @@ end = struct
       | Pexp_setfield (_, _, e0) when e0 == exp -> Some (LessMinus, Non)
       | Pexp_setinstvar _ -> Some (LessMinus, Non)
       | Pexp_field _ -> Some (Dot, Left)
-      | Pexp_send _ -> Some (HashOp, Non)
+      | Pexp_send _ -> Some (Dot, Left)
```
Comment (Julow) [inline src/Ast.ml]: You should add a comment to say it's not how it is in the parser and that it is intended.

**Comment by gpetiot** [comment]
> The `Dot` precedence seems to work.
> 
> This two lines below have the same AST, should we add parenthesis for disambiguation ?
> 
> ```ocaml
> (a #+ b)#m #= c;;
> a #+ b#m #= c;;
> ```

These two lines both format to `(a#+b)#m#=c` on this branch

File: src/Ast.ml (hunk 2 above)
Comment (gpetiot) [reply src/Ast.ml]: done

**Pull request merged**